pkg/registry/apis/example/register.go

```diff
@@ -33,39 +33,69 @@ var _ grafanaapiserver.APIGroupBuilder = (*TestingAPIBuilder)(nil)
 // This is used just so wire has something unique to return
 type TestingAPIBuilder struct {
 	codecs serializer.CodecFactory
+	gv     schema.GroupVersion
 }
 
 func RegisterAPIService(features featuremgmt.FeatureToggles, apiregistration grafanaapiserver.APIRegistrar) *TestingAPIBuilder {
 	if !features.IsEnabled(featuremgmt.FlagGrafanaAPIServerWithExperimentalAPIs) {
 		return nil // skip registration unless opting into experimental apis
 	}
-	builder := &TestingAPIBuilder{}
+	builder := &TestingAPIBuilder{
+		gv: schema.GroupVersion{Group: GroupName, Version: VersionID},
+	}
 	apiregistration.RegisterAPI(builder)
 	return builder
 }
 
 func (b *TestingAPIBuilder) GetGroupVersion() schema.GroupVersion {
-	return SchemeGroupVersion
+	return b.gv
+}
+
+func addKnownTypes(scheme *runtime.Scheme, gv schema.GroupVersion) {
+	scheme.AddKnownTypes(gv,
+		&example.RuntimeInfo{},
+		&example.DummyResource{},
+		&example.DummyResourceList{},
+		&example.DummySubresource{},
+	)
 }
 
 func (b *TestingAPIBuilder) InstallSchema(scheme *runtime.Scheme) error {
-	err := AddToScheme(scheme)
-	if err != nil {
-		return err
-	}
-	return scheme.SetVersionPriority(SchemeGroupVersion)
+	addKnownTypes(scheme, b.gv)
+
+	// Link this version to the internal representation.
+	// This is used for server-side-apply (PATCH), and avoids the error:
+	//   "no kind is registered for the type"
+	addKnownTypes(scheme, schema.GroupVersion{
+		Group:   b.gv.Group,
+		Version: runtime.APIVersionInternal,
+	})
+
+	// If multiple versions exist, then register conversions from zz_generated.conversion.go
+	// if err := playlist.RegisterConversions(scheme); err != nil {
+	//   return err
+	// }
+	metav1.AddToGroupVersion(scheme, b.gv)
+	return scheme.SetVersionPriority(b.gv)
 }
 
 func (b *TestingAPIBuilder) GetAPIGroupInfo(
 	scheme *runtime.Scheme,
 	codecs serializer.CodecFactory, // pointer?
 	optsGetter generic.RESTOptionsGetter,
 ) (*genericapiserver.APIGroupInfo, error) {
+	b.codecs = codecs
 	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(GroupName, scheme, metav1.ParameterCodec, codecs)
+
+	runtimeStorage := newDeploymentInfoStorage(b.gv, scheme)
+	dummyStorage := newDummyStorage(b.gv, scheme,
+		"test1", "test2", "test3")
+
```

Comment:
I'm not sure that I understand why we don't initialize `dummySubStorage := &DummySubresourceREST{Store: dummyStorage.Store}` here rather than inside `dummyStorage`, it seems like we can decouple things since dummyStorage itself doesn't need to know anything about the subresource?

Reply:
I cleaned this up after revisiting some of my k8s assumptions.  I thought this required a public `Store *genericregistry.Store` property to match both the parent+child.... but turns out that is not required :)